pkg/repository/local_git.go

```diff
@@ -189,3 +194,29 @@ func (lg *LocalGit) GetLatestCommit(ctx context.Context, gitPath, dirPath string
 	}
 	return string(output), nil
 }
+
+// GetLatestCommit returns the latest commit of the repository at the given path
+func (lg *LocalGit) Diff(ctx context.Context, gitPath, dirPath string, fixAttempt int) (string, error) {
```

Comment:
can we add tests for this func?

Reply:
Yes, this PR needs to revisit all the test. I wanted to have all the merge conflicts before start creating the rest of the tests